src/main/scala/ai/privado/utility/Utilities.scala

```diff
@@ -62,6 +66,20 @@ object Utilities {
 
   private val logger = LoggerFactory.getLogger(getClass)
 
+  def getEngineContext(maxCallDepthP: Int = 4)(implicit semanticsP: Semantics = DefaultSemantics()): EngineContext = {
+    val expanLimit =
+      if ScanProcessor.config.limitArgExpansionDataflows > -1 then ScanProcessor.config.limitArgExpansionDataflows
```

Comment:
Instead of using the `ScanProcessor.config` as a static variable, better to pass it in method `getEngineContext`, as it con create conflicting result in writing testcases, due to ScanProcessor.config holding a single value

Reply:
done